mono/metadata/metadata.c

```diff
@@ -3016,12 +2982,11 @@ mono_metadata_lookup_generic_class (MonoClass *container_class, MonoGenericInst
 	MonoImageSet *set;
 	CollectData data;
 
+	memset (&helper, 0, sizeof(helper)); // act like g_new0
```

Comment:
I do think this should stay-- there are two problems here:

1. This code sets NULL all fields needed by `collect_gclass_images`, but this is essentially a coincidence. Someone who edited `collect_gclass_images` would not know that they could cause a crash by using new fields-- the contract that the function only uses those fields are used is totally implicit.

2. Passing around partially-invalid objects complicates debugging. If I remember correctly this one came up for me because I was looking at the object after it had escaped this function, I tried looking at the owner so I could try to get a name, owner was set to a non-NULL object and I wound up getting gibberish. This lead to some wasted time trying to figure out if there was heap corruption before realizing I had just read an uninitialized pointer, since we're so consistent about initializing entire structures to NULL elsewhere…

I think initializing to zero is consistent with other uses of this key-object-on-the-stack.